streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -583,6 +627,172 @@ public void advanceWallClockTime(final long advanceMs) {
+     * @param valueSerializer the Serializer for the value type
+     * @param <K> the key type
+     * @param <V> the value type
+     * @return {@link TestInputTopic} object
+     */
+    public final <K, V> TestInputTopic<K, V> createInputTopic(final String topicName,
+                                                              final Serializer<K> keySerializer,
+                                                              final Serializer<V> valueSerializer) {
+        return new TestInputTopic<K, V>(this, topicName, keySerializer, valueSerializer);
+    }
+
+    /**
+     * Create {@link TestInputTopic} to be used for piping records to topic
+     * Uses provided start timestamp and autoAdvance parameter for records
+     *
+     * @param topicName             the name of the topic
+     * @param keySerializer   the Serializer for the key type
+     * @param valueSerializer the Serializer for the value type
+     * @param startTimestamp Start timestamp for auto-generated record time
+     * @param autoAdvance autoAdvance duration for auto-generated record time
+     * @param <K> the key type
+     * @param <V> the value type
+     * @return {@link TestInputTopic} object
+     */
+    public final <K, V> TestInputTopic<K, V> createInputTopic(final String topicName,
+                                                              final Serializer<K> keySerializer,
+                                                              final Serializer<V> valueSerializer,
+                                                              final Instant startTimestamp,
+                                                              final Duration autoAdvance) {
+        return new TestInputTopic<K, V>(this, topicName, keySerializer, valueSerializer, startTimestamp, autoAdvance);
+    }
+
+    /**
+     * Create {@link TestOutputTopic} to be used for reading records from topic
+     *
+     * @param topicName             the name of the topic
+     * @param keyDeserializer   the Deserializer for the key type
+     * @param valueDeserializer the Deserializer for the value type
+     * @param <K> the key type
+     * @param <V> the value type
+     * @return {@link TestOutputTopic} object
+     */
+    public final <K, V> TestOutputTopic<K, V> createOutputTopic(final String topicName,
+                                                                final Deserializer<K> keyDeserializer,
+                                                                final Deserializer<V> valueDeserializer) {
+        return new TestOutputTopic<K, V>(this, topicName, keyDeserializer, valueDeserializer);
+    }
+
+    /**
+     * Read the next record from the given topic.
+     * These records were output by the topology during the previous calls to {@link #pipeInput(ConsumerRecord)}.
+     *
+     * @param topic the name of the topic
+     * @return the next record on that topic, or {@code null} if there is no record available
+     */
+    @SuppressWarnings("WeakerAccess")
```

Comment:
I'm curious why these suppressions were necessary. I'd have thought that the package-private members would all have been here because they were needed by the I/O Topic classes.

Is it because IDEA thinks that protected is "stronger" than package-private?

Reply:
Not sure the reason.